config-model/src/main/java/com/yahoo/vespa/model/application/validation/Validation.java

```diff
@@ -99,9 +105,17 @@
                 new ContainerRestartValidator(),
                 new NodeResourceChangeValidator()
         };
-        return Arrays.stream(validators)
-                .flatMap(v -> v.validate(currentModel, nextModel, overrides, now).stream())
-                .collect(toList());
+        List<ConfigChangeAction> actions = Arrays.stream(validators)
+                                                 .flatMap(v -> v.validate(currentModel, nextModel, overrides, now).stream())
+                                                 .collect(toList());
+        overrides.invalid(actions.stream()
+                                 .filter(DisallowableConfigChangeAction.class::isInstance)
```

Comment:
Given this usage, I think it would be simpler to just put validationId on ConfigChangeAction as an Optional.

Reply:
I suppose that's equally not super-obvious, but good enough, and a bit simpler, yes. Can't even remember why I didn't do this. Probably no good reason. It's "you must provide a validatoin id for implementations which may block deployment", in any case.

Reply:
Ah, so it's because making it optional makes you have to `.get()` or `.orElseThrow()` to get the `name()` in the child interfaces, which is ugly. It's clearer with a separate interface, although more verbose.

Reply:
Should just remove the `allowed()` altogether, since it no longer serves a purpose with these changes. Guess there'' be some more work here, then ...